systemd-swap

```diff
@@ -318,6 +318,29 @@ case "$1" in
     fi
 
     if YN "${swapfc_enabled}"; then
+      # validate swapfc_frequency due to possible issues caused if set incorrectly
+      if [[ -n "${swapfc_frequency//[0-9]}" || -z "$swapfc_frequency" ]]; then
+        ERRO "swapfc_frequency is not set to a valid integer value: ${swapfc_frequency}"
+      elif (( swapfc_frequency < 1 || swapfc_frequency > 86400 )); then
+        WARN "swapfc_frequency must be in range of 1..86400: ${swapfc_frequency} - set to 1"
+        swapfc_frequency=1
```

Comment:
A lower limit of 0 would essentially pin a cpu core/thread to 100% - I don't know any scenario that would be useful in.

Reply:
Well the limit would be `< 0`, essentially we've supported setting `swapfc_frequency` to say 0.5 and I don't see a reason not to continue to do that.

Reply:
If you want to allow that kind of precision, all right, I'll change it.

Reply:
Since I cannot do easy floating point maths or tests within bash built-ins, this is going to be ugly though.

Reply:
Won't `swapfc_frequency < 0` just work?

Reply:
The sanity check only accepts integers. If it is adjusted for floats, every arithmetic expression involving `swapfc_frequency` won't work anymore since bash doesn't support floating point math.

You basically need to do the calculations with an external tool like `bc`.

I don't think I'm going to change this code since it is way too ugly in my opinion. Just close the PR if  you want to keep compatibility with decimals for the frequency.